OpenXava/src/org/openxava/web/editors/AmazonS3Persistor.java

```diff
@@ -0,0 +1,151 @@
+	  
+   public static String amazonBucketName() {
+	   /** Amazon s3 Bucket Name */
+	   return NaviOXPreferences.getInstance().getAmazonBucketName();
+   
+   }
+	  
+   public static void uploadToAmazonS3(String filename) {
+	   try {
+       	   
+		   /** 
+		    * Initializing the Amazon S3 Client 
+		    * */
+		   AmazonS3Client s3 = initialize();
+		   
+		   /** Amazon S3 Bucket name */
+	       String bucketName = amazonBucketName();
+           
+	       /**
+	        *  Uploading a new object to S3 from a file..
+	        *  1) Creating a file object using the uploaded filename
+	        *  2) Uploading the file to s3 with bucketname, filename as key and file object. 
+	        */
+	       
+	       File file = new File( PARENT +"/"+ filename.toString());
+
+	       System.out.println("Uploading a new object to S3 from a file" + file.getName());         
+	       
+	       s3.putObject(new PutObjectRequest(bucketName, PrefixFolderName+file.getName(), file));
+           
+           /** Deleting the new file stored in local directory */
+           FileUtils.deleteQuietly(file);
+           
+       } catch (AmazonServiceException ase) {
+    	   
+    	   amazonException(ase);
+       
+       } 
+	   	catch (AmazonClientException ace) {
+    	   
+	   		amazonClientException(ace);
+       
+	   	} 
+   }
+   
+   
+   public static void downloadFromAmazonS3(String uuid) throws IOException {
+	   try 
+		{
+		    /** Initializing the Amazon S3 Client Connection with access key and secret key */
+			AmazonS3Client s3 = initialize();
+			
+			/** Amazon S3 Bucket name */
+	        String bucketName = amazonBucketName();
+	         
+	        /** 
+	         * Fetching all to objects with prefix with unique uuid 
+	         * Note: Always only one object will be present in  
+	         */
+	        
+			ObjectListing s3Objects = s3.listObjects(bucketName, PrefixFolderName+uuid);
+			
+			/**
+		     *  Downloading the file from s3..
+		     *  1) Loop through the Objectlisting and get key to get the object.
+		     *  2) Copy the content to local directory. 
+		     */
+			
+			for (S3ObjectSummary objectSummary: s3Objects.getObjectSummaries()) {
+			
+				String key = objectSummary.getKey();
+		        
+				S3Object s3object = s3.getObject(new GetObjectRequest(bucketName, key));
+				
+				S3ObjectInputStream inputStream = ((S3Object) s3object).getObjectContent();
+			    
+				String keyWithoutPrefix = (PrefixFolderName.isEmpty()) ? key : key.split(PrefixFolderName)[1];		
+				
+				FileUtils.copyInputStreamToFile(inputStream, new File(PARENT+"/"+keyWithoutPrefix));
+    			
+			}
+		} catch (AmazonServiceException ase) {
+			
+			amazonException(ase);
+        
+		} catch (AmazonClientException ace) {
+    	   
+        	amazonClientException(ace);
+        }  
+   }
+   
+   public static void amazonException(AmazonServiceException ase) {
+	   System.out.println("Caught an AmazonServiceException, which means your request made it "
```

Comment:
Question: How does OX typically log messages? If there is a logger system this code should follow the same approach